From 1297a852f1008d57ea397fd4ea09e3769004e7ee Mon Sep 17 00:00:00 2001
From: Jonathan Scruggs <j.scruggs@gmail.com>
Date: Tue, 5 Sep 2017 12:17:31 +0100
Subject: [PATCH] Use PkgConfig to find IlmBase and OpenEXR

Those two programs have .pc files so use those instead of the
complicated Find files. The .pc files take into account the naming
of those libraries.

Signed-off by: Jonathan Scruggs <j.scruggs@gmail.com>
---
 cmake/FindILMBase.cmake | 187 ------------------------------------------------
 cmake/FindOpenEXR.cmake | 125 --------------------------------
 openvdb/CMakeLists.txt  |  18 ++---
 3 files changed, 9 insertions(+), 321 deletions(-)
 delete mode 100644 cmake/FindILMBase.cmake
 delete mode 100644 cmake/FindOpenEXR.cmake

diff --git a/cmake/FindILMBase.cmake b/cmake/FindILMBase.cmake
deleted file mode 100644
index b803b87..0000000
--- a/cmake/FindILMBase.cmake
+++ /dev/null
@@ -1,187 +0,0 @@
-# Copyright (c) 2012-2016 DreamWorks Animation LLC
-#
-# All rights reserved. This software is distributed under the
-# Mozilla Public License 2.0 ( http://www.mozilla.org/MPL/2.0/ )
-#
-# Redistributions of source code must retain the above copyright
-# and license notice and the following restrictions and disclaimer.
-#
-# *     Neither the name of DreamWorks Animation nor the names of
-# its contributors may be used to endorse or promote products derived
-# from this software without specific prior written permission.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY INDIRECT, INCIDENTAL,
-# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-# IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
-# LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
-#
-
-#-*-cmake-*-
-# - Find ILMBase
-#
-# Author : Nicholas Yue yue.nicholas@gmail.com
-#
-# This auxiliary CMake file helps in find the ILMBASE headers and libraries
-#
-# ILMBASE_FOUND                  set if ILMBASE is found.
-# ILMBASE_INCLUDE_DIR            ILMBASE's include directory
-# ILMBASE_LIBRARYDIR            ILMBASE's include directory
-# Ilmbase_HALF_LIBRARY           ILMBASE's Half libraries
-# Ilmbase_IEX_LIBRARY            ILMBASE's Iex libraries
-# Ilmbase_IEXMATH_LIBRARY        ILMBASE's IexMath libraries
-# Ilmbase_ILMTHREAD_LIBRARY      ILMBASE's IlmThread libraries
-# Ilmbase_IMATH_LIBRARY          ILMBASE's Imath libraries
-
-FIND_PACKAGE ( PackageHandleStandardArgs )
-
-FIND_PATH ( ILMBASE_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h
-  "$ENV{ILMBASE_ROOT}"
-  NO_DEFAULT_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH
-  )
-
-FIND_PACKAGE_HANDLE_STANDARD_ARGS ( ILMBase
-  REQUIRED_VARS ILMBASE_LOCATION
-  )
-
-OPTION ( ILMBASE_NAMESPACE_VERSIONING "Namespace versioning of libraries" ON )
-
-IF ( ILMBASE_FOUND )
-
-  FILE ( STRINGS "${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h" _ilmbase_version_major_string REGEX "#define ILMBASE_VERSION_MAJOR ")
-  STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MAJOR" "" _ilmbase_version_major_unstrip "${_ilmbase_version_major_string}")
-  STRING ( STRIP "${_ilmbase_version_major_unstrip}" ILMBASE_VERSION_MAJOR )
-
-  FILE ( STRINGS "${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h" _ilmbase_version_minor_string REGEX "#define ILMBASE_VERSION_MINOR ")
-  STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MINOR" "" _ilmbase_version_minor_unstrip "${_ilmbase_version_minor_string}")
-  STRING ( STRIP "${_ilmbase_version_minor_unstrip}" ILMBASE_VERSION_MINOR )
-
-  IF ( ILMBASE_NAMESPACE_VERSIONING )
-	SET ( IEX_LIBRARY_NAME       Iex-${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR}       )
-	SET ( IEXMATH_LIBRARY_NAME   IexMath-${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR}   )
-	SET ( ILMTHREAD_LIBRARY_NAME IlmThread-${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR} )
-	SET ( IMATH_LIBRARY_NAME     Imath-${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR}     )
-  ELSE ( ILMBASE_NAMESPACE_VERSIONING )
-	SET ( IEX_LIBRARY_NAME       Iex       )
-	SET ( IEXMATH_LIBRARY_NAME   IexMath   )
-	SET ( ILMTHREAD_LIBRARY_NAME IlmThread )
-	SET ( IMATH_LIBRARY_NAME     Imath     )
-  ENDIF ( ILMBASE_NAMESPACE_VERSIONING )
-
-  SET ( ILMBASE_INCLUDE_DIRS
-    ${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}
-    ${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR
-    CACHE STRING "ILMBase include directories")
-  SET ( ILMBASE_LIBRARYDIR ${ILMBASE_LOCATION}/${CMAKE_INSTALL_LIBDIR}
-    CACHE STRING "ILMBase library directories")
-  SET ( ILMBASE_FOUND TRUE )
-  
-  SET ( ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
-  IF (Ilmbase_USE_STATIC_LIBS)
-    IF (APPLE)
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY Iex PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY IlmThread PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY Imath PATHS ${ILMBASE_LIBRARYDIR} )
-    ELSEIF (WIN32)
-      # Link library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half_static PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY Iex_static PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY IlmThread_static PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY Imath_static PATHS ${ILMBASE_LIBRARYDIR} )
-    ELSE (APPLE)
-      SET ( CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		)
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY Iex PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		)
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY IlmThread PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		)
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY Imath PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		)
-    ENDIF (APPLE)
-  ELSE (Ilmbase_USE_STATIC_LIBS)
-    IF (APPLE)
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib")
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY Iex PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY IlmThread PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY Imath PATHS ${ILMBASE_LIBRARYDIR} )
-    ELSEIF (WIN32)
-      # Link library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY ${IEX_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IEXMATH_LIBRARY ${IEXMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY ${ILMTHREAD_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR} )
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY ${IMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR} )
-      # Load library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
-      FIND_LIBRARY ( Ilmbase_HALF_DLL Half PATHS ${ILMBASE_LOCATION}/bin
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IEX_DLL ${IEX_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IEXMATH_DLL ${IEXMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_DLL ${ILMTHREAD_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IMATH_DLL ${IMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-    ELSE (APPLE)
-      FIND_LIBRARY ( Ilmbase_HALF_LIBRARY Half PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IEX_LIBRARY ${IEX_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IEXMATH_LIBRARY ${IEXMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_ILMTHREAD_LIBRARY ${ILMTHREAD_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-      FIND_LIBRARY ( Ilmbase_IMATH_LIBRARY ${IMATH_LIBRARY_NAME} PATHS ${ILMBASE_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-        )
-    ENDIF (APPLE)
-  ENDIF ()
-  # MUST reset
-  SET(CMAKE_FIND_LIBRARY_SUFFIXES ${ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES})
-
-ELSE ( ILMBASE_FOUND )
-  MESSAGE ( FATAL_ERROR "Unable to find ILMBase, ILMBASE_ROOT = $ENV{ILMBASE_ROOT}")
-ENDIF ( ILMBASE_FOUND )
diff --git a/cmake/FindOpenEXR.cmake b/cmake/FindOpenEXR.cmake
deleted file mode 100644
index 1fa4875..0000000
--- a/cmake/FindOpenEXR.cmake
+++ /dev/null
@@ -1,125 +0,0 @@
-# Copyright (c) 2012-2016 DreamWorks Animation LLC
-#
-# All rights reserved. This software is distributed under the
-# Mozilla Public License 2.0 ( http://www.mozilla.org/MPL/2.0/ )
-#
-# Redistributions of source code must retain the above copyright
-# and license notice and the following restrictions and disclaimer.
-#
-# *     Neither the name of DreamWorks Animation nor the names of
-# its contributors may be used to endorse or promote products derived
-# from this software without specific prior written permission.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY INDIRECT, INCIDENTAL,
-# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-# IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
-# LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
-#
-
-# -*- cmake -*-
-# - Find OpenEXR
-#
-# Author : Nicholas Yue yue.nicholas@gmail.com
-#
-# This module will define the following variables:
-#  OPENEXR_INCLUDE_DIRS - Location of the openexr includes
-#  OPENEXR_LIBRARIES - [TODO] Required libraries for all requested bindings
-#  OPENEXR_FOUND - true if OPENEXR was found on the system
-#  OPENEXR_LIBRARYDIR - the full set of library directories
-
-FIND_PACKAGE ( PackageHandleStandardArgs )
-
-FIND_PATH ( OPENEXR_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h
-  ENV OPENEXR_ROOT
-  NO_DEFAULT_PATH
-  NO_SYSTEM_ENVIRONMENT_PATH
-  )
-
-FIND_PACKAGE_HANDLE_STANDARD_ARGS ( OpenEXR
-  REQUIRED_VARS OPENEXR_LOCATION
-  )
-
-OPTION ( OPENEXR_NAMESPACE_VERSIONING "Namespace versioning of libraries" ON )
-
-IF ( OPENEXR_FOUND )
-
-  FILE ( STRINGS "${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h" _openexr_version_major_string REGEX "#define OPENEXR_VERSION_MAJOR ")
-  STRING ( REGEX REPLACE "#define OPENEXR_VERSION_MAJOR" "" _openexr_version_major_unstrip "${_openexr_version_major_string}")
-  STRING ( STRIP "${_openexr_version_major_unstrip}" OPENEXR_VERSION_MAJOR )
-
-  FILE ( STRINGS "${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h" _openexr_version_minor_string REGEX "#define OPENEXR_VERSION_MINOR ")
-  STRING ( REGEX REPLACE "#define OPENEXR_VERSION_MINOR" "" _openexr_version_minor_unstrip "${_openexr_version_minor_string}")
-  STRING ( STRIP "${_openexr_version_minor_unstrip}" OPENEXR_VERSION_MINOR )
-  
-  MESSAGE ( STATUS "Found OpenEXR v${OPENEXR_VERSION_MAJOR}.${OPENEXR_VERSION_MINOR} at ${OPENEXR_LOCATION}" )
-
-  IF ( OPENEXR_NAMESPACE_VERSIONING )
-	SET ( ILMIMF_LIBRARY_NAME IlmImf-${OPENEXR_VERSION_MAJOR}_${OPENEXR_VERSION_MINOR} )
-  ELSE ( OPENEXR_NAMESPACE_VERSIONING )
-	SET ( ILMIMF_LIBRARY_NAME IlmImf )
-  ENDIF ( OPENEXR_NAMESPACE_VERSIONING )
-	
-  SET ( OPENEXR_INCLUDE_DIRS
-    ${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}
-    ${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR
-    CACHE STRING "Openexr include directories")
-  SET ( OPENEXR_LIBRARYDIR ${OPENEXR_LOCATION}/${CMAKE_INSTALL_LIBDIR}
-    CACHE STRING "Openexr library directories")
-  SET ( OPENEXR_FOUND TRUE )
-
-
-  IF (Openexr_USE_STATIC_LIBS)
-    IF (APPLE)
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR} )
-    ELSEIF (WIN32)
-      # Link library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR} )
-    ELSE (APPLE)
-      # MESSAGE ( "CMAKE_FIND_LIBRARY_SUFFIXES = " ${CMAKE_FIND_LIBRARY_SUFFIXES})
-      SET ( ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
-      SET ( CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		)
-      SET ( CMAKE_FIND_LIBRARY_SUFFIXES ${ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES})
-    ENDIF (APPLE)
-  ELSE ()
-    IF (APPLE)
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib")
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR} )
-    ELSEIF (WIN32)
-      # Link library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR} )
-      # Load library
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
-      FIND_LIBRARY ( Openexr_ILMIMF_DLL ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LOCATION}/bin )
-      # MUST reset
-      SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-    ELSE (APPLE)
-      FIND_LIBRARY ( Openexr_ILMIMF_LIBRARY ${ILMIMF_LIBRARY_NAME} PATHS ${OPENEXR_LIBRARYDIR}
-		NO_DEFAULT_PATH
-		NO_CMAKE_ENVIRONMENT_PATH
-		NO_CMAKE_PATH
-		NO_SYSTEM_ENVIRONMENT_PATH
-		NO_CMAKE_SYSTEM_PATH
-		)
-    ENDIF (APPLE)
-  ENDIF ()
-
-  # SET( Openexr_ILMIMF_LIBRARY ${OPENEXR_ILMIMF_LIBRARY_PATH} CACHE STRING "Openexr's IlmImf library")
-  
-ENDIF ( OPENEXR_FOUND )
-
diff --git a/openvdb/CMakeLists.txt b/openvdb/CMakeLists.txt
index 790f627..4f1ad8f 100644
--- a/openvdb/CMakeLists.txt
+++ b/openvdb/CMakeLists.txt
@@ -80,11 +80,12 @@ IF (WIN32 AND OPENVDB_DISABLE_BOOST_IMPLICIT_LINKING)
   ADD_DEFINITIONS ( -DBOOST_ALL_NO_LIB )
 ENDIF ()
 
+FIND_PACKAGE ( PkgConfig )
 FIND_PACKAGE ( Blosc REQUIRED )
 FIND_PACKAGE ( TBB REQUIRED )
 FIND_PACKAGE ( ZLIB REQUIRED )
-FIND_PACKAGE ( ILMBase REQUIRED )
-FIND_PACKAGE ( OpenEXR REQUIRED )
+PKG_CHECK_MODULES ( ILMBASE_LIBS REQUIRED IlmBase )
+PKG_CHECK_MODULES ( OPENEXR_LIBS REQUIRED OpenEXR )
 FIND_PACKAGE ( OpenGL REQUIRED )
 IF (OPENVDB_BUILD_DOCS)
   FIND_PACKAGE ( Doxygen REQUIRED )
@@ -108,8 +109,8 @@ IF ( NOT OPENVDB_BUILD_HOUDINI_SOPS )
   INCLUDE_DIRECTORIES ( SYSTEM ${ZLIB_INCLUDE_DIR} )
 ENDIF ()
 
-INCLUDE_DIRECTORIES ( SYSTEM ${ILMBASE_INCLUDE_DIRS} )
-INCLUDE_DIRECTORIES ( SYSTEM ${OPENEXR_INCLUDE_DIRS} )
+INCLUDE_DIRECTORIES ( SYSTEM ${ILMBASE_LIBS_INCLUDE_DIRS} )
+INCLUDE_DIRECTORIES ( SYSTEM ${OPENEXR_LIBS_INCLUDE_DIRS} )
 INCLUDE_DIRECTORIES ( SYSTEM ${GLFW_INCLUDE_DIRECTORY} )
 
 IF (WIN32)
@@ -181,7 +182,7 @@ TARGET_LINK_LIBRARIES ( openvdb_static
   ${Boost_IOSTREAMS_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
   ${Tbb_TBB_LIBRARY}
-  ${Ilmbase_HALF_LIBRARY}
+  ${ILMBASE_LIBS_LDFLAGS}
   ${ZLIB_LIBRARY}
   ${BLOSC_blosc_LIBRARY}
   )
@@ -189,7 +190,7 @@ TARGET_LINK_LIBRARIES ( openvdb_shared
   ${Boost_IOSTREAMS_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
   ${Tbb_TBB_LIBRARY}
-  ${Ilmbase_HALF_LIBRARY}
+  ${ILMBASE_LIBS_LDFLAGS}
   ${ZLIB_LIBRARY}
   ${BLOSC_blosc_LIBRARY}
   )
@@ -241,9 +242,8 @@ TARGET_LINK_LIBRARIES ( vdb_render
   openvdb_shared
   ${CMAKE_THREAD_LIBS_INIT}
   ${BLOSC_blosc_LIBRARY}
-  ${Openexr_ILMIMF_LIBRARY}
-  ${Ilmbase_ILMTHREAD_LIBRARY}
-  ${Ilmbase_IEX_LIBRARY}
+  ${OPENEXR_LIBS_LDFLAGS}
+  ${ILMBASE_LIBS_LDFLAGS}
   )
 
 SET ( VDB_VIEW_SOURCE_FILES
-- 
2.14.1

